src/omnisharp/constants/IMonoResolver.ts

```diff
@@ -0,0 +1,11 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Options } from "../options";
+import { MonoInformation } from "./MonoInformation";
+
+export interface IMonoResolver {
+    shouldUseGlobalMono(options: Options): Promise<MonoInformation>;
```

Comment:
"should" in the name makes me think this returns a boolean. Maybe call this "getMonoInfo" or somethign?